selfdrive/car/tests/test_fw_fingerprint.py

```diff
@@ -104,5 +115,27 @@ def test_fw_request_ecu_whitelist(self):
                          f'{brand.title()}: FW query whitelist missing ecus: {ecu_strings}')
 
 
+class TestFwFingerprintTiming(unittest.TestCase):
+  @parameterized.expand([(1,), (2,)])
+  def test_fw_query_timing(self, num_pandas):
+    params = Params()
+
+    for brand, config in FW_QUERY_CONFIGS.items():
+      with self.subTest(brand=brand, num_pandas=num_pandas):
+        fake_socket = FakeSocket()
+        thread = threading.Thread(target=get_fw_versions, args=(fake_socket, fake_socket, brand), kwargs=dict(num_pandas=num_pandas))
+        thread.start()
+
+        t = time.perf_counter()
+        while thread.is_alive():
+          time.sleep(0.01)
+          if not params.get_bool("ObdMultiplexingChanged"):
+            params.put_bool("ObdMultiplexingChanged", True)
+
+        total_time = round(time.perf_counter() - t, 2)
+        self.assertLessEqual(total_time, 1.3)
```

Comment:
let's add a min time too, so that we have to update the test thresholds if it improves

Reply:
the thresholds should also be brand specific

Reply:
done, added the theoretical calculated time with some tolerance